starfish/recipe/cli.py

```diff
@@ -0,0 +1,29 @@
+from slicedimage.io import resolve_path_or_url
+
+from starfish.config import StarfishConfig
+from starfish.starfish import starfish
+from starfish.util import click
+from .recipe import Recipe
+
+
+@starfish.command("recipe")
+@click.option("--recipe", required=True, type=str, metavar="RECIPE_PATH_OR_URL")
+@click.option(
+    "--input", type=str, multiple=True, metavar="INPUT_FILE_PATH_OR_URL",
+    help="input file paths or urls to map to the recipe input parameters")
+@click.option(
+    "--output", type=str, multiple=True, metavar="OUTPUT_FILE_PATH",
+    help="output file paths to write recipe outputs to")
+@click.pass_context
+def run_recipe(ctx, recipe, input, output):
+    """Runs a recipe with a given set of inputs and outputs."""
+    config = StarfishConfig()
+
+    backend, relativeurl, _ = resolve_path_or_url(
```

Comment:
General style question: We have a bunch of custom classes, and in this case I needed to follow 3 wrapper functions  in slicedimage.io to find out that this `backend` is a subclass of `slicedimage.backends.Backend`. 

Since we have type hints, I would find it helpful to use them in the cases of our custom classes, so that developers have a better sense of what the code is doing. 

In this case: 

```python
    backend: slicedimage.io.Backend
    relativeurl: str
    backend, relativeurl, _ = resolve_path_or_url(
        recipe, backend_config=config.slicedimage)


```

Reply:
Sure, I can do this.

Reply:
Correction: `Backend` was never exposed.

I will correct this.

Reply:
https://github.com/spacetx/starfish/issues/1246